pkg/ruler/ruler.go

```diff
@@ -1061,11 +1061,12 @@ func (r *Ruler) ruleGroupListToGroupStateDesc(userID string, backupGroups rulesp
 
 		groupDesc := &GroupStateDesc{
 			Group: &rulespb.RuleGroupDesc{
-				Name:      group.GetName(),
-				Namespace: group.GetNamespace(),
-				Interval:  interval,
-				User:      userID,
-				Limit:     group.Limit,
+				Name:        group.GetName(),
+				Namespace:   group.GetNamespace(),
+				Interval:    interval,
+				User:        userID,
+				Limit:       group.Limit,
+				QueryOffset: group.QueryOffset,
 			},
 			// We are keeping default value for EvaluationTimestamp and EvaluationDuration since the backup is not evaluating
```

Comment:
@yeya24 I also wasn't sure about this change, but added the new field for consistency. That said, the comment below this says:

> We are keeping default value for EvaluationTimestamp and EvaluationDuration since the backup is not evaluating

So maybe it deserves to be treated like EvaluationTimestamp and EvaluationDuration for the purpose of backups? If that's the case I can revert this change and update the comment.

Reply:
If I understand correctly, the backup stores the rule inmemory to serve List rules API requests only. Since query offset is not part of the response then we probably don't need to store that. I am also fine to store it incase it is used in the future.

@rajagopalanand or @rapphil please correct me if I was wrong.

Reply:
It seems we also use the same path to reply to rpc requests which will export the entire obj. This would add it to the rpc response.
https://github.com/cortexproject/cortex/blob/master/pkg/ruler/ruler.go#L1230

I am not sure how well used is the rpc api, but maybe we want to maintain both the same.

Reply:
Thanks for the additional info. So it sounds like this change is fine to leave as is?

Reply:
> It seems we also use the same path to reply to rpc requests which will export the entire obj

This RPC seems only being used when listing rules, which doesn't need this field.
But let's leave it as is. There is another field used as response https://github.com/cortexproject/cortex/blob/master/pkg/ruler/api.go#L185 so we are good.